### ISSUE TITLE
Create new development branches

### ISSUE THREAD

**Issue by fpetrogalli-arm** [shibatch/sleef#49]
As mentioned in #48, we should consider using multiple branches for our development.

The current situation is:

1. `master` -> new features development and releases
2. `cmake-transition` -> port of the makefile build system to `cmake`

Once 2 is done, we can merge `cmake-transition` into master.

@shibatch , can you explain how you would like to change this? Having different branches for different version seems like to be a good idea. What would end up in the 3.1.* branch and in teh 3.2 branch? how they would relate to `master` and `cmake-transition`?

One alternative solution would be to keep master being the release branch, and `git tag` those points in the history of `master` where we do a particular release, and consider any other branch a development branch that we might merge into master when ready, and mark with a release tag if needed.

**Comment by shibatch** [comment]
For the 3.1.* branch, we will make minor bug fixes only.
Main development will take place on the master branch, and cmake-transition branch will be merged to the master branch eventually, but not to 3.1.* branch.

If we use the alternative solution, we cannot merge the development branch to the master branch before, in our case, release of 3.2. Before that, there is always possibility to release a new 3.1.*. I think this is not convenient.

**Comment by fpetrogalli-arm** [comment]
Thanks for teh explanation. Now I see your point, yes, a 3.1.* branch for bug fixes is a good idea.

**Comment by fpetrogalli-arm** [comment]
Closing this, we moved to cmake long ago :)
